### DISCUSSION TITLE
Sorting `Secret`s

### DISCUSSION THREAD

**Discussion by huyparody** [sbertix/Swiftagram#254] [Ideas]
Hi @sbertix 

I tried to login 3 different account, and use `Authenticator.userDefaults.secrets.get()` to get all the Secrets, but everytime I fetch, the array of Secrets change the order, can you reorder it with login timestamp?
 e.g. account 1 login first so the order of that Secret would be at `[0]`.

**Comment by sbertix** [comment]
Hey @huyparody, 

It's something that has already been considered in the past, tbh, but I never went through with it cause I assumed users would rely on something a bit more robust than just a cookie timestamp sorting (which is beyond the scopes of the library, though). 
I'm not against it though, so it's something that can totally be considered. The only thing I'd probably do different is having the last logged in `Secret` first, but again, just an opinion.

**Reply by huyparody** [reply]
> Hey @huyparody,
> 
> It's something that has already been considered in the past, tbh, but I never went through with it cause I assumed users would rely on something a bit more robust than just a cookie timestamp sorting (which is beyond the scopes of the library, though).
> I'm not against it though, so it's something that can totally be considered. The only thing I'd probably do different is having the last logged in `Secret` first, but again, just an opinion.

Do you have any idea, beside "having the last logged in `Secret`", and how should I have it, saves it to UserDefaults or something?

**Reply by sbertix** [reply]
I mean, if you need the feature right now, I suggest storing either the custom order yourself, for instance — as you mentioned — through `UserDefaults`, or just applying a custom sorting every time you call `Authenticator.userDefaults.secrets.get()`. 

I'm okay with trying to add some kind of sorting in the next release, though, whatever it might end up being.

**Reply by sbertix** [reply]
Of course when I say "storing the custom order yourself", I mean just the identifiers. So you could save them in `UserDefaults` and then map the array to a `Secret` one using the `Authenticator` accessory.
